docs/guide/en/field-configuration.md

```diff
@@ -1,33 +1,175 @@
+where:
 
-## Common configuration from theme (optional)
+- key is a theme name;
+- value is a mapping (associative array) between [settings'](#theme) names and their corresponding values.
 
-TODO: describe
+```php
+use Yiisoft\Form\Theme\ThemeContainer;
 
-## Individual configuration from theme (optional);
+ThemeContainer::initialize(
+    configs: [
+        'main' => [ 
+            'containerClass' => 'field-container-main',
+            // ...            
+            'fieldConfigs' => [
+                Checkbox::class => [
+                    'inputContainerTag()' => ['div'],
+                    // ...
+                ],
+                // ...
+            ],
+        ],
+        'alternative' => [
+            'containerClass' => 'field-container-alt',
+            // ...            
+            'fieldConfigs' => [
+                Checkbox::class => [
+                    'inputContainerTag()' => ['span'],
+                    // ...
+                ],
+                // ...
+            ],
+        ],      
+    ],
+    defaultConfig: 'main',
+    validationRulesEnricher: new MyValidationRulesEnricher(),
+);
+```
 
-TODO: describe
+You can additionally set (optional):
 
-## configuration at the place of use
+- config used as a default one using `defaultConfig` option;
+- [enricher](validation-rules-enrichment.md) instance used to enrich fields depending on their type from given 
+validation rules.
 
-TODO: add example of config passed through widget() call
-    
-    
+## Built-in themes
+
+These themes are available out of the box:
+
+- Bootstrap 5 Horizontal;
+- Bootstrap 5 Vertical.
+
+Their settings are stored in separate configuration files. To simplify including it, you can use 
+`Yiisoft\Form\Theme\ThemePath` - constants with for all built-in themes.
+
+```php
+use Yiisoft\Form\Theme\ThemeContainer;
+use Yiisoft\Form\Theme\ThemePath;
+
+ThemeContainer::initialize(
+    config: [
+        'vertical' => require ThemePath::BOOTSTRAP5_VERTICAL,
+        'horizontal' => require ThemePath::BOOTSTRAP5_HORIZONTAL,
+    ],
+    defaultConfig: 'vertical',
+);
+```
+
+## Using with Yii Config
+
+When using [Yii Config](https://github.com/yiisoft/config), there is no need to manually interact with theme cotnainer,
+it's initialized automatically during application's bootstrap. Here is an example of relevant `config/params.php` file
+section:
+
+```php
+use Yiisoft\Form\Theme\ThemePath;
+
+return [
+    // ...
+    'yiisoft/form' => [
+        'themes' => [
+            'vertical' => require ThemePath::BOOTSTRAP5_VERTICAL,
+            'horizontal' => require ThemePath::BOOTSTRAP5_HORIZONTAL,
+        ],
+        'defaultTheme' => 'vertical',
+        'validationRulesEnricher' => new MyValidationRulesEnricher(),
+    ],
+    // ...
+];
+```
+
+No built-in themes and validation rules enricher are used by default.
+
+## Themes' preview
```

Comment:
Move this to internals.

Reply:
Why? Is preview for developers only? I think it can be useful for users too.

Reply:
Yes. It is for developers only.

Reply:
OK. I moved it to internals.